src/modules/fancyzones/editor/FancyZonesEditor/EditorOverlay.xaml.cs

```diff
@@ -18,8 +18,11 @@ public partial class EditorOverlay : Window
 
         private readonly Settings _settings = ((App)Application.Current).ZoneSettings;
         private LayoutPreview _layoutPreview;
+
         private UserControl _editor;
 
+        private static MainWindow _mainWindow = new MainWindow();

```

Comment:
Is there any reason for changes in this file?

Reply:
This is one of the things we need to solve better.. E.g. think of some communication protocol and which communication channel to be used between editor and engine.

At the moment, editor and engine communicates via tmp .json files:
 - When starting the editor,  engine checks tmp files for possible data sent from editor last time editor was alive[1], process them, and then creates tmp .json files to pass necessary data to editor
 - editor creates tmp file when on Closing()

So, originally, every time we click `Edit Selected Layout` Editor MainWindow was closed, and if we just close (click X) EditorWindow we are back to newly created MainWindow. With this change and changes in MainWindow.xaml.cs (Close() -> Hide()) we now have single MainWindow instance which is just hidden and shown when needed. This way communication explained above is possible.. 
E.g. 
1) start editor
2) delete one of the custom zone sets
3) go Edit Selected Layout
4) close it (click X)
5) we are back to editor MainWindow and deleted custom zone set is still there because engine didn't process data sent from editor on Closing() because it would read the data on next editor spawning (see [1])